search_engine_parser/core/engines/google.py

```diff
@@ -87,7 +92,19 @@ def parse_single_result(self, single_result, return_type=ReturnType.FULL, **kwar
                 link_tag = desc_tag.find('a')
                 if link_tag:
                     raw_link = link_tag.get('href')
+                    results['hrefs'] = self.extract_href(raw_link)
                     results['links'] = self.parse_url(raw_link)
             desc = desc_tag.text
             results['descriptions'] = desc
         return results
+
+    def extract_href(self, url):
```

Comment:
this should be `clean_url` we are not extracting the url alone, we are processing it.

Reply:
In the docstring add so that doctest can pick it up when running
```python
""" extract clean URL from the extracted URL

> clean_url("https://www.google.com/url?q=https://english.stackexchange.com/questions/140710/what-is-the-opposite-of-preaching-to-the-choir&sa=U&ved=2ahUKEwi31MGyzvnuAhXyyDgGHXXACOYQFnoECAkQAg&usg=AOvVaw1GdXON-JIWGu-dGjHfgljl")
https://english.stackexchange.com/questions/140710/what-is-the-opposite-of-preaching-to-the-choir
"""
```

Reply:
Remember to call it from to the `parse_url` function

https://github.com/bisoncorps/search-engine-parser/blob/1e44f194e9b61375809ed8635c1498b979c87830/search_engine_parser/core/engines/google.py#L37

should be 
```python
return clean_url(urljoin(self.base_url, url))
```

Reply:
@MeNsaaH I had created a different field in the GoogleSearch object, to store the cleaned URLs.
The reason is that `clean_url` may fail for many reasons, and it that case we would loose the original URL if we overwrite the same object attribute.
What do you recommend?

Reply:
We can add a `raw_url` field for that. However, links should return the cleaned urls. or What do you think? @deven96

Reply:
Yes that's actually what I thought when I found this package: I thought the url were already cleaned.
You are proposing to rename the current `links` to `raw_urls` and then to use the `clean_url` to make the new `links`. Is my understanding correct? I think it makes sense. @MeNsaaH

Reply:
Exactly @pgrandinetti